openedx/core/djangoapps/user_authn/api/views.py

```diff
@@ -61,20 +78,26 @@ def get(self, request, **kwargs):  # lint-amnesty, pylint: disable=unused-argume
             },
         }
 
-        if settings.ENABLE_DYNAMIC_REGISTRATION_FIELDS and is_register_page:
-            registration_fields = RegistrationFieldsContext().get_fields()
-            context['registration_fields'].update({
-                'fields': registration_fields,
-            })
-
-            optional_fields = RegistrationFieldsContext('optional').get_fields()
-            if optional_fields:
-                extended_profile_fields = configuration_helpers.get_value('extended_profile_fields')
-                context['optional_fields'].update({
-                    'fields': optional_fields,
-                    'extended_profile': extended_profile_fields if extended_profile_fields else []
+        if settings.ENABLE_DYNAMIC_REGISTRATION_FIELDS:
+            optional_fields = self._get_optional_fields_context()
```

Comment:
The implementation before this (having `is_register_page`) was also optimizing the code by not getting optional fields for login page context. 
I know we have 3 pages getting context now but the current implementation will also get optional fields for the login page which is not required.

Reply:
It won't return it in the MFE context but yes you are right it will still call this function. Let me update it.